tree_in_context/src/utility/util.egg

```diff
@@ -35,3 +35,37 @@
        (= len1 (tuple-length expr1))
        (= ith (tuple-ith expr2 i)))
        ((union (tuple-ith (Concat ord expr1 expr2) (+ len1 i)) ith)) :ruleset always-run)
+
+(function tuple-remove-fst (Expr) Expr :unextractable)
+(rule ((tuple-remove-fst tup)
+       (= tup (Concat ord (Single expr1) expr2)))
```

Comment:
Is it possible that the first argument to `Concat` is `(Let ... (Single ...))`?

Reply:
I have the same question about  `tuple-ith`. We probably had the assumption that  a tuple is always a tree of uninterrupted `Concat`s and `Single`s. Would be nice to enforce that in our type system, or leave a note somewhere.